src/cpu/aarch64/matmul/acl_matmul.hpp

```diff
@@ -0,0 +1,155 @@
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef ACL_MATMUL_HPP
+#define ACL_MATMUL_HPP
+
+#include "cpu/aarch64/matmul/acl_matmul_utils.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+namespace matmul {
+
+using namespace dnnl::impl::cpu::matmul;
+
+struct acl_resource_t : public resource_t {
+    acl_resource_t() : acl_obj_(utils::make_unique<acl_matmul_obj_t>()) {}
+
+    status_t configure(const acl_matmul_conf_t &amp) {
+        if (!acl_obj_) return status::out_of_memory;
+        acl_obj_->src_tensor.allocator()->init(amp.src_info);
+        acl_obj_->wei_tensor.allocator()->init(amp.wei_info);
+        acl_obj_->dst_tensor.allocator()->init(amp.dst_info);
+        // Configure transpose kernel for src, wei or both
+        if (amp.is_transA) {
+            acl_obj_->src_acc_tensor.allocator()->init(amp.src_acc_info);
+            acl_obj_->transA.configure(
+                    &acl_obj_->src_acc_tensor, &acl_obj_->src_tensor);
+        }
+        if (amp.is_transB) {
+            acl_obj_->wei_acc_tensor.allocator()->init(amp.wei_acc_info);
+            acl_obj_->transB.configure(
+                    &acl_obj_->wei_acc_tensor, &acl_obj_->wei_tensor);
+        }
+        // Configure GEMM
+        acl_obj_->gemm.configure(&acl_obj_->src_tensor, &acl_obj_->wei_tensor,
+                nullptr, &acl_obj_->dst_tensor, amp.alpha, 0.0f, amp.gemm_info);
+        return status::success;
+    }
+    acl_matmul_obj_t &get_acl_obj() const { return *acl_obj_; }
+
+    DNNL_DISALLOW_COPY_AND_ASSIGN(acl_resource_t);
+
+private:
+    std::unique_ptr<acl_matmul_obj_t> acl_obj_;
+};
+
+struct acl_matmul_t : public primitive_t {
+    struct pd_t : public cpu_matmul_pd_t {
+
+        pd_t(const matmul_desc_t *adesc, const primitive_attr_t *attr,
+                const cpu_matmul_pd_t *hint_fwd_pd)
+            : cpu_matmul_pd_t(adesc, attr, hint_fwd_pd), amp_() {}
+
+        using cpu_matmul_pd_t::cpu_matmul_pd_t;
+
+        DECLARE_COMMON_PD_T("matmul:acl", acl_matmul_t, USE_GLOBAL_SCRATCHPAD);
+
+        status_t init(engine_t *engine) {
+            using smask_t = primitive_attr_t::skip_mask_t;
+            bool ok = src_md()->data_type == data_type::f32
+                    && weights_md()->data_type == data_type::f32
+                    && desc()->accum_data_type == data_type::f32
+                    && dst_md()->data_type == data_type::f32
+                    && platform::has_data_type_support(data_type::f32)
+                    && attr()->has_default_values(
+                            smask_t::oscale_runtime | smask_t::post_ops)
```

Comment:
flag `oscale_runtime` means that output scales can be either provided at primitive creation time in attributes or at execution time as an additional argument to the primitive. However during execution alpha is taken from attributes:
```cpp
// Set alpha (output scaling)
amp.alpha = attr.output_scales_.scales_[0];
```
This can be fixed by:
- Option A: prohibit `oscale_runtime` and allow `oscale` only
- Option B: Add a dispatching mechanism to get alpha from either attributes or arguments based on the value in attributes. In case of `oscale_runtime` oscale value in attributes will be equal to `DNNL_RUNTIME_F32_VAL`, which means that the scales will be provided at execution time.

Reply:
Thanks. I will make this change to option A.